components/automate-cli/cmd/chef-automate/cmdUtils.go

```diff
@@ -73,132 +80,131 @@ func (c *remoteCmdExecutor) Execute() error {
+		return map[string][]CmdResult{}, errors.New("Missing or Unsupported flag")
 	}
-
-	return nil
 }
 
-// executeCmdOnGivenNodes executes given command/commands on all given nodes concurrently.
-func (c *remoteCmdExecutor) executeCmdOnGivenNodes(input *CmdInputs, nodeIps []string, remoteService string, timestamp string, cliWriter *cli.Writer) {
-	resultChan := make(chan CmdResult, len(nodeIps))
+func (c *remoteCmdExecutor) executeCmdOnGivenNodes(input *CmdInputs, nodeIps []string, remoteService string, timestamp string, cliWriter *cli.Writer) map[string][]CmdResult {
+	command := make(map[string]string)
+	ouputJsonResultMapMutex := sync.RWMutex{}
 	originalSSHConfig := c.SshUtil.getSSHConfig()
 	inputFiles := input.InputFiles
 	outputFiles := input.Outputfiles
-	command := input.Cmd
+	if len(input.MutipleCmdWithArgs) == 0 {
+		command["Script"] = input.Cmd
+	} else {
+		command = input.MutipleCmdWithArgs
+	}
 	timeout := input.WaitTimeout
-
 	inputFileToOutputFileMap := map[string]string{}
 	for _, file := range inputFiles {
 		destinationFile := remoteService + "_" + timestamp + "_" + file
 		inputFileToOutputFileMap[file] = destinationFile
 	}
-
-	for _, hostIP := range nodeIps {
-		newSSHConfig := &SSHConfig{
-			sshUser:    originalSSHConfig.sshUser,
-			sshPort:    originalSSHConfig.sshPort,
-			sshKeyFile: originalSSHConfig.sshKeyFile,
-			hostIP:     hostIP,
-			timeout:    timeout,
-		}
-		sshUtil := NewSSHUtil(newSSHConfig)
-
-		printConnectionMessage(remoteService, hostIP, cliWriter)
-
-		go c.executeCmdOnNode(command, inputFileToOutputFileMap, outputFiles, remoteService, input.ErrorCheckEnableInOutput, sshUtil, resultChan)
+	ouputJsonResult := make(map[string][]CmdResult)
+	var wg sync.WaitGroup
+	for scriptName, cmdScript := range command {
+
+		wg.Add(1)
+		go func(originalSSHConfig *SSHConfig, timeout int, cmdScript string, scriptName string, inputFileToOutputFileMap map[string]string, outputFiles []string, remoteService string, errorCheckEnableInOutput bool) {
+			defer wg.Done()
+			for _, hostIP := range nodeIps {
+				sshConfig := &SSHConfig{
+					sshUser:    originalSSHConfig.sshUser,
+					sshPort:    originalSSHConfig.sshPort,
+					sshKeyFile: originalSSHConfig.sshKeyFile,
+					hostIP:     hostIP,
+					timeout:    timeout,
+				}
+				if !input.HideSSHConnectionMessage {
+					printConnectionMessage(remoteService, hostIP, cliWriter)
+				}
+				result := c.executeCmdOnNode(cmdScript, scriptName, inputFileToOutputFileMap, outputFiles, remoteService, errorCheckEnableInOutput, sshConfig)
+				ouputJsonResultMapMutex.Lock()
+				ouputJsonResult[result.HostIP] = append(ouputJsonResult[result.HostIP], result)
+				ouputJsonResultMapMutex.Unlock()
+				if !input.SkipPrintOutput {
+					cliWriter.Println("=====================================================")
+					printOutput(remoteService, result, outputFiles, c.Output)
+					cliWriter.Println("=====================================================")
```

Comment:
put all this code in a single method preferably inside printOutput method

Reply:
Done